packages/rum-core/src/common/service-factory.js

```diff
@@ -32,22 +32,23 @@ import {
   LOGGING_SERVICE,
   APM_SERVER
 } from './constants'
+import { __DEV__ } from '../env'
+
+const serviceCreators = {
+  [CONFIG_SERVICE]: () => new ConfigService(),
+  [LOGGING_SERVICE]: () =>
+    new LoggingService({
+      prefix: '[Elastic APM] '
+    }),
+  [APM_SERVER]: factory => {
+    return new ApmServer(
+      ...factory.getService([CONFIG_SERVICE, LOGGING_SERVICE])
+    )
+  }
+}
 
 class ServiceFactory {
   constructor() {
-    this._creators = {
```

Comment:
Having the creators here is actually better and we are modifying the creators in another files seems like a sideeffect. Lets use the same patter and instead reduce the naming of instance methods to something like `addCreator` and `addInstance` instead of big `registerServiceInstance` etc.

Reply:
I compared the size reduction for having creators on the class, it nearly had 0.06 KB reduction in size, so I kept it this way. 

You're right about the side-effect but I think we probably should accept some side-effect in favour of the bundle size reduction.